docs/reference/language/context.md

```diff
@@ -25,38 +25,201 @@ in some places that are also aware of their location in the document:
+Read 2 still outputs `40pt`, because `text.size` is a constant.
+However, this output is printed in 25pt font, as specified by the set
+rule before the read. This illustrates the importance of picking the 
+right insertion point for a context to get access to precisely the right 
+styles. If you need access to updated property fields after a set rule, 
+you can use _nested contexts_:
+
+```example
+#set text(lang: "en")
+#context [
+  Read 1: #text.lang
+
+  #set text(lang: "fr")
+  Read 2: #context text.lang
+]
+```
+
+All of the above applies to `show` rules analogously. To demonstrate this, 
+we define a function `template` which is activated by an "everything" show 
+rule in a context:
+
+```example
+#let template(body) = {
+  set text(size: 25pt)
+  body
+}
+
+#set text(size: 40pt)
+#context [
+  Read 1: #text.size
+
+  #show: template
+  Read 2: #text.size \
+  Read 3: #context text.size
+]
+```
+Reads 1 and 2 print the original text size upon entry in the first 
+context (since `text.size` remains constant there), but Read 3 is 
+located in a nested context and reflects the new font size set by 
+the `show` rule via the `template` function.
+
+## Using context-dependent property fields to control content appearance
+An important purpose of reading the current value of properties is, 
+of course, to use this information in the calculation of derived 
+properties, instead of setting those properties manually. For example, 
+you can double the font size like this:
+
+```example
+#context [
+  // the context allows you to
+  // retrieve the current text.size
+  #set text(size: text.size * 200%)
+  Large text \ 
+]
+Original size
+```
+
+Since set rules are only active until the end of the enclosing scope, 
+"Original size" is printed with the original font size.
+For the specific case of accessing `text.size`, context is usually not necessary as the `1em` unit is always equal to the current font size, so the above example is equivalent to 
+
+```example
+#[
+  #set text(size: 2em)
+  Large text \ 
+]
+Original size
+```
+
+but convenient alternatives like this are unavailable for most properties.
+This makes contexts a powerful and versatile concept. For example, 
+you can use a similar resizing technique to increase the spacing 
+between the lines of a specific equation block (or any other content):
+
+```example
+#let spaced(spacing: 100%, body) = context {
+  // access current par.leading in a context
+  set par(leading: par.leading * spacing)
+  body
+}
+
+Normal spacing:
+$ x \ x $
+Doubled spacing:
+#spaced(spacing: 200%)[$ z \ z $]
+```
+
+The advantage of this technique is that the user does not have to know the 
+original spacing in order to double it. To double the spacing of all 
+equations, you can put the same calculations in a show rule. Note that 
+it is not necessary to add the `context` keyword on the right-hand side 
+of a `show` rule, because show rules establish a context automatically:
```

Comment:
The fact that show rules establish context automatically is maybe a bit too buried. It's quite important I think.